server/game/core/gameSteps/phases/ActionPhase.ts

```diff
@@ -35,7 +35,15 @@ export class ActionPhase extends Phase {
 
     private rotateActiveQueueNextAction() {
         // breaks the action loop if both players have passed
-        this.game.queueSimpleStep(() => this.game.rotateActivePlayer(), 'rotateActivePlayer');
+        this.game.queueSimpleStep(() => {
+            const activePlayer = this.game.getActivePlayer();
+            if (activePlayer && activePlayer.hasOngoingEffect(EffectName.AdditionalAction)) {
+                activePlayer.removeOngoingEffects(EffectName.AdditionalAction);
+            } else {
+                this.game.rotateActivePlayer();
+            }
+        }, 'rotateActivePlayer');
```

Comment:
This is a naive implementation that only supports a single additional action. The `additionalAction` effect already supports an `amount`, so we could reduce by 1 here instead, but we don't have a case for that yet.